pkg/tracker/gas.go

```diff
@@ -36,18 +37,17 @@ func (b *GasTracker) String() string {
 	return "GasTracker"
 }
 
-func NewGasTracker(logger log.Logger) *GasTracker {
+func NewGasTracker(logger log.Logger, db db.DB, client rpc.ETHClient) *GasTracker {
 	return &GasTracker{
+		db:     db,
+		client: client,
 		logger: log.With(logger, "component", "gas tracker"),
 	}
 
 }
 
 func (b *GasTracker) Exec(ctx context.Context) error {
-	client := ctx.Value(tellorCommon.ClientContextKey).(rpc.ETHClient)
-	DB := ctx.Value(tellorCommon.DBContextKey).(db.DB)
-
-	netID, err := client.NetworkID(context.Background())
+	netID, err := b.client.NetworkID(context.Background())
```

Comment:
Here and in a few other places when we have an instance of the context should use it instead of creating a new one.
```suggestion
	netID, err := b.client.NetworkID(ctx)
```